controller/controlplane/controller_watch.go

```diff
@@ -146,23 +146,48 @@ func (r *Reconciler) getControlPlaneRequestFromManagedByNameLabel(ctx context.Co
 		return
 	}
 
-	for _, controlplane := range controlplanes.Items {
-		// For cluster-scoped resources, the owner is set through a label.
-		if obj.GetLabels()[consts.GatewayOperatorManagedByNameLabel] == controlplane.Name {
-			return []reconcile.Request{
-				{
-					NamespacedName: types.NamespacedName{
-						Namespace: controlplane.Namespace,
-						Name:      controlplane.Name,
-					},
+	for i := range controlplanes.Items {
+		if !objectIsOwnedByControlPlane(obj, &controlplanes.Items[i]) {
+			continue
+		}
+
+		return []reconcile.Request{
+			{
+				NamespacedName: types.NamespacedName{
+					Namespace: controlplanes.Items[i].Namespace,
+					Name:      controlplanes.Items[i].Name,
 				},
-			}
+			},
 		}
 	}
 
 	return
 }
 
+// objectIsOwnedByControlPlane checks if the object is owned by the control plane.
+//
+// NOTE: We are using the managed-by-name label to identify the owner of the resource
+// To keep backward compatibility, we also check the owner reference which
+// are not used anymore for cluster-scoped resources since that's considered
+// an error.
+func objectIsOwnedByControlPlane(obj client.Object, cp *operatorv1beta1.ControlPlane) bool {
+	if k8sutils.IsOwnedByRefUID(obj, cp.GetUID()) {
+		return true
+	}
+
+	if labels := obj.GetLabels(); len(labels) > 0 {
+		if labels[consts.GatewayOperatorManagedByNameLabel] == cp.Name {
+			if obj.GetNamespace() != "" {
+				return cp.GetNamespace() == obj.GetNamespace()
+			} else {
+				return true
+			}
+		}
+	}
```

Comment:
```suggestion
	if labels[consts.GatewayOperatorManagedByNameLabel] == cp.Name {
		if obj.GetNamespace() != "" {
			return cp.GetNamespace() == obj.GetNamespace()
		} else {
			return true
		}
	}
```

Reading from empty/nil map is safe it returns default object value in this case empty string - `""`. The same as for non-existing item. So the first check can be dropped without changing semantic of this snippet